panic in the TestColumnAdd

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x73c51f6]
goroutine 39441 [running]:
github.com/pingcap/tidb/pkg/ddl_test.checkAddPublic({0x9d50a08, 0xc00881bef0}, {0x0, 0x0}, {0x9d0ce28?, 0xc007461ce0?})
	pkg/ddl/column_change_test.go:316 +0x1d6
github.com/pingcap/tidb/pkg/ddl_test.TestColumnAdd.func1(0xc00851fb00)
	pkg/ddl/column_change_test.go:80 +0x286
github.com/pingcap/tidb/pkg/ddl/util/callback.(*TestDDLCallback).OnJobUpdated(0xc0087789a0, 0xc00851fb00?)
	pkg/ddl/util/callback/callback.go:124 +0x1dd
github.com/pingcap/tidb/pkg/ddl.(*ddl).delivery2worker.func1()
	pkg/ddl/job_table.go:439 +0x171b
github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).Run.func1()
	pkg/util/wait_group_wrapper.go:157 +0x87
created by github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).Run in goroutine 38134
	pkg/util/wait_group_wrapper.go:155 +0xe5 
```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/ddl/util/callback/callback.go
+++ b/pkg/ddl/util/callback/callback.go
@@ -124,6 +124,9 @@ func (tc *TestDDLCallback) OnJobUpdated(job *model.Job) {
 		(*onJobUpdatedExportedFunc)(job)
 		return
 	}
+	if job.State == model.JobStateSynced {
+		return
+	}
 	if tc.onJobUpdated != nil {
 		tc.onJobUpdated(job)
 		return

